Mathlib/RingTheory/PrincipalIdealDomainOfPrime.lean

```diff
@@ -3,63 +3,63 @@ Copyright (c) 2018 Chris Hughes. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Anne Baanen
 -/
-import Mathlib.RingTheory.Ideal.Colon
+import Mathlib.RingTheory.Ideal.Oka
 import Mathlib.RingTheory.PrincipalIdealDomain
 
 /-!
 # Principal ideal domains and prime ideals
 
-# Main results
+## Main results
 
-- `IsPrincipalIdeal.of_prime`: a ring where all prime ideals are principal is a principal ideal ring
+- `IsPrincipalIdealRing.of_prime`: a ring where all prime ideals are principal is a principal ideal
+  ring.
 -/
 
-open Ideal
+variable {R : Type*} [CommSemiring R]
+
+namespace Ideal
+
+/-- If a commutative (semi)ring has a non-principal ideal, then it has an ideal which is maximal for
+not being principal. -/
+lemma exists_maximal_not_principal (h : ∃ I : Ideal R, ¬I.IsPrincipal) :
+    ∃ I : Ideal R, Maximal (¬·.IsPrincipal) I := by
```

Comment:
So this is the second time I'm seeing this pattern (the first time is #28451), so I'm wondering if we can have an auxiliary lemma similar to [Ideal.IsOka.forall_of_forall_prime](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Ideal/Oka.html#Ideal.IsOka.forall_of_forall_prime), but instead the condition `hmax : (∃ I, ¬P I) → ∃ I, Maximal (fun x => ¬P x) I` would be instead replaced with "for all nonempty chain of ideals, if their union satisfies P, then one of the elements satisfy P".